server/controllers/request.js

```diff
@@ -0,0 +1,99 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+
+/*
+@route GET /requests/getRequests
+@desc for fetching all the requests for a particular user
+@access private
+*/
+exports.getRequests = asyncHandler( async( req, res, next ) => {
+    const  user_id  = req.user.id;
```

Comment:
I noticed that we are supposed to store the user's profile id in the request document. Maybe search for a user and then search for a request using their profile id? 

Note: Maybe consider changing the name of the field in your model from 'user_id' to something like 'owner_id'